src/builtins.jl

```diff
@@ -152,11 +152,7 @@ function maybe_evaluate_builtin(frame, call_expr, expand::Bool)
             return Some{Any}(Core.finalizer(getargs(args, frame)...))
         end
     elseif @static isdefined(Core, :get_binding_type) && f === Core.get_binding_type
-        if nargs == 2
-            return Some{Any}(Core.get_binding_type(@lookup(frame, args[2]), @lookup(frame, args[3])))
-        else
-            return Some{Any}(Core.get_binding_type(getargs(args, frame)...))
-        end
+        return Some{Any}(Core.get_binding_type(getargs(args, frame)...))
```

Comment:
I thought splatting here was bad for inference, which is why `n == 2` was special cased before?

Reply:
Now the compiler has the special casings for these builtin functions and it doesn't register any tfuncs for them. Maybe we can recover tfuncs that are simplified version of the special cases.